packages/scandipwa/src/component/CategoryConfigurableAttributes/CategoryConfigurableAttributes.component.js

```diff
@@ -119,6 +119,17 @@ export class CategoryConfigurableAttributes extends ProductConfigurableAttribute
         switch (attribute_code) {
         case 'price':
             return this.renderPriceSwatch(option);
+        case 'category_id':
+            const { childrenCategories } = this.props;
+            const { attribute_values } = option;
+            const childrenCategoryIds = childrenCategories.map((category) => category.id.toString());
+            const subCategoriesIds = attribute_values.filter((item) => childrenCategoryIds.includes(item));
+            if (subCategoriesIds.length) {
+                this.props.configurable_options.category_id.attribute_values = subCategoriesIds;
```

Comment:
Please prefer destructuring assignment over the using long chains of properties.
And don't forget about the default values in case if some chain part is unexpectedly undefined.
This is better
```javascript
const {
  prop: {
    propAttr: {
      val
    } = {}
  } = {}
} = this.props;
```
than this
```javascript
const val = this.props.prop.propAttr.val;
```

Reply:
Also please move this code to a separate method, like it is in the `price` and `default` sections.
This improves code readability and class extensibility.